Private repository access?

## Situation
oauth2 client/secret created in either org or user
org-owned repo selected in frontend

## Expected outcome
can see milestones

## Actual outcome
Appears to be 404 (github unauthorized)
```
2021-11-17T17:53:08.350261+00:00 app[web.1]: Caused by: java.io.FileNotFoundException: https://api.github.com/repos/FATMAP/aws-dns-zones
2021-11-17T17:53:08.350261+00:00 app[web.1]: 	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1893) ~[na:1.8.0_312-heroku]
2021-11-17T17:53:08.350261+00:00 app[web.1]: 	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1495) ~[na:1.8.0_312-heroku]
2021-11-17T17:53:08.350261+00:00 app[web.1]: 	at java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:480) ~[na:1.8.0_312-heroku]
2021-11-17T17:53:08.350262+00:00 app[web.1]: 	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:352) ~[na:1.8.0_312-heroku]
2021-11-17T17:53:08.350262+00:00 app[web.1]: 	at org.kohsuke.github.GitHubHttpUrlConnectionClient.getResponseInfo(GitHubHttpUrlConnectionClient.java:69) ~[github-api-1.111.jar!/:na]
2021-11-17T17:53:08.350262+00:00 app[web.1]: 	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:348) ~[github-api-1.111.jar!/:na]
2021-11-17T17:53:08.350262+00:00 app[web.1]: 	... 108 common frames omitted
2021-11-17T17:53:08.350263+00:00 app[web.1]: 
2021-11-17T17:53:08.357167+00:00 heroku[router]: at=info method=GET path="/milestones/[redacted]/aws-dns-zones" host=sre-stoneboard.herokuapp.com request_id=c89dc471-77a5-473b-8924-356f93aff6b6 fwd="94.139.27.210" dyno=web.1 connect=0ms service=231ms status=500 bytes=701 protocol=https
```

## More info

If I use a public repo, it works fine. I guess that the oauth callback needs to request additional scopes, probably by first getting the orgs and then prompting the user to select which, if any, orgs to grant access to.

## Additional question

If/when I get this working, can I see milestones from multiple repos at once? Project + talk really resonates with me, thanks for the hard work

--- a/README.md
+++ b/README.md
@@ -59,6 +59,7 @@ Config variables for local `.env` and Heroku app:
 ```
 GITHUB_CLIENT_ID=A-GITHUB-CLIENT-ID
 GITHUB_CLIENT_SECRET=A-GITHUB-CLIENT-SECRET
+GITHUB_CLIENT_SCOPES=user # set this to user,repo if you need to access private repositories
 GITHUB_HOSTNAME=github.com (optional, used for GitHub Enterprise custom domains)
 REPOSITORIES=org/repo,org/repo...
 ```

--- a/src/main/resources/application.yml
+++ b/src/main/resources/application.yml
@@ -13,6 +13,7 @@ spring:
       client:
         registration:
           github:
+            scope: ${GITHUB_CLIENT_SCOPES}
             clientId: ${GITHUB_CLIENT_ID}
             clientSecret: ${GITHUB_CLIENT_SECRET}
 ---
@@ -28,6 +29,7 @@ spring:
             clientSecret: ${GITHUB_CLIENT_SECRET}
         provider:
           github:
+            scope: ${GITHUB_CLIENT_SCOPES}
             authorization-uri: https://${GITHUB_HOSTNAME}/login/oauth/authorize
             token-uri: https://${GITHUB_HOSTNAME}/login/oauth/access_token
             user-info-uri: https://${GITHUB_HOSTNAME}/api/v3/user

